pallets/data-preservers/src/lib.rs

```diff
@@ -62,6 +63,144 @@ use {
 pub mod pallet {
     use super::*;
 
+    #[cfg(feature = "std")]
+    pub trait SerdeIfStd: serde::Serialize + serde::de::DeserializeOwned {}
```

Comment:
I've seen this in polkadot-sdk, can we use that one instead?

https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_frame/traits/trait.MaybeSerialize.html

Reply:
Tried to replace it with `MaybeSerializeDeserialize` but I can't get it to build :/ When creating the types with `cfg_attr(feature = "std", ...)` it tells me Serialize/Deserialize is not implemented (even without std enabled). I'd keep the current code for now and we can do a cleanup PR later (with other code cleanups, such as using `macro_rules_attribute` to remove all those annoyingly long derives)

Reply:
Ok no worries

Reply:
For the `GenesisConfig` types need to implement serde traits even in no-std. However it seems it is no longer needed to feature-gate serde in our crate. We should then in a later PR cleanup serde usages to remove the feature-gates.